### PR TITLE
Add babeld support to ubus-lime-metrics and last_internet_path

### PR DESCRIPTION
**Pull request by gmarcos87**
Currently lime-ubus tools are tied to bmx6, this pull request adds support for babeld detecting which of the two protocols is installed.

### CODE REVIEW

File: packages/ubus-lime-metrics/files/etc/uci-defaults/99-save-last-known-path-to-internet [hunk 1]
```diff
@@ -1,5 +1,4 @@
 #!/bin/sh
 echo "adding cron to save last known path to the Internet"
-sed "\| \* \* \* .*last_internet_path.*|d" -i /etc/crontabs/root
-echo "*/30 * * * * /usr/sbin/internet_path > /tmp/last_internet; if [ -s /tmp/last_internet ] && [\ "$(md5sum /tmp/last_internet | awk '{ print $1 }')\" != \"$(md5sum /etc/last_internet |  awk '{ print $1 }
-')\" ]; then cp /tmp/last_internet /etc/last_internet; fi" >>  /etc/crontabs/root
+sed "\| \* \* \* .*internet_path*|d" -i /etc/crontabs/root
+echo "*/30 * * * * /usr/sbin/internet_path save" >>  /etc/crontabs/root
```
Comment (G10h4ck) [inline packages/ubus-lime-metrics/files/etc/uci-defaults/99-save-last-known-path-to-internet]: Don't you prefere the unique_append idiom, lime in other packages?

File: packages/ubus-lime-metrics/files/usr/libexec/rpcd/lime-metrics [hunk 2]
```diff
@@ -105,6 +126,21 @@ local function get_gateway(msg)
    printJson(_get_gateway())
 end
 
+local function get_last_internet_path(msg)
+    local path = list_from_file("/etc/last_internet_path")
```
Comment (G10h4ck) [inline packages/ubus-lime-metrics/files/usr/libexec/rpcd/lime-metrics]: This information looks like quite volatile, and to store it in persistent flash memory cold cause flash breakage to many write cycles, is it necessary to store this information persistently?

File: packages/ubus-lime-metrics/files/usr/sbin/last_internet [hunk 3]
```diff
@@ -2,15 +2,44 @@
...
-# if 8.8.8.8 is accessible, save internet path to file
-if ping -q -c5 -w10 8.8.8.8 &>/dev/null
-    then path=`mtr -6 -r -c 1 $gw.mesh  | grep "\.|" |  awk '{ print $2}' | cut -d'.' -f1`
-    printf "$path"
+if [ ! -z "$result" ] && [ "$1" = "save" ] ; then
+    newSum=$(echo $result | md5sum | cut -d' ' -f 1)
+    oldSum=$(md5sum /etc/last_internet_path | cut -d' ' -f 1)
+    if [ "$newSum" != "$oldSum" ]; then
+        echo -n -e "$result" > /etc/last_internet_path
+        echo 'Last path to internet updated'
```
Comment (G10h4ck) [inline packages/ubus-lime-metrics/files/usr/sbin/last_internet]: Probably we should monitor how often this happens, to determine how much flash write cycles are we burning with this

Comment (gmarcos87) [reply packages/ubus-lime-metrics/files/usr/sbin/last_internet]: In a stable network that path does not change regularly, we could avoid overwriting in another way: perform metrics and save only if they are better, or simply save only the shortest path if it is different from the previous one.

File: packages/ubus-lime-metrics/files/usr/libexec/rpcd/lime-metrics (hunk 2 above)
Comment (gmarcos87) [reply packages/ubus-lime-metrics/files/usr/libexec/rpcd/lime-metrics]: It is not really volatile information, it is information that must persist when the router restarts. Suppose that the user loses the Internet connection. The first thing they do is to restart the router, or if an electrical outage occurs in part of the network, etc. In order to be able to diagnose what happens with your usual exit to interent you need to be able to remember where you used to go.

File: packages/ubus-lime-metrics/files/etc/uci-defaults/99-save-last-known-path-to-internet (hunk 1 above)
Comment (gmarcos87) [reply packages/ubus-lime-metrics/files/etc/uci-defaults/99-save-last-known-path-to-internet]: Yeah, this package is older and I just fixed the script. I can adapt it to the unique_append style

Comment (G10h4ck) [reply packages/ubus-lime-metrics/files/etc/uci-defaults/99-save-last-known-path-to-internet]: better use the unique_append (is this the exact name?) idiom then ;)

**Pull request merged**